### PR TITLE
feat: add fast trigonometric functions

### PR DESCRIPTION
**Pull request by Soptq**
This PR adds fast trigonometric functions (sin, cos, tan) to the library. Compared to standard trigonometric functions, fast trigonometric functions are computational and spacial efficient, with minor (acceptable errors).

## Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

- Add fast trigonometric functions (sin, cos, tan) to the library
- All tests are passed

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/math/src/trigonometry.cairo [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+// Since there is no float in cairo, we multiply every number by 1e8
+// # Arguments
+// * `x` - The number to calculate cos(x)
+// # Returns
+// * `bool` - true if the result is positive, false if the result is negative
+// * `u64` - cos(x) * 1e8
+// # Example
+// * fast_cos(6000000000) = (true, 50000000)
+fn fast_cos(x: u64) -> (bool, u64) {
+    let multipier = 100000000_u64;
```
Comment (LucasLvy) [inline src/math/src/trigonometry.cairo:68]: use `consteval_int`

File: src/math/src/trigonometry.cairo [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+// Calculate fast tan(x)
+// Since there is no float in cairo, we multiply every number by 1e8
+// # Arguments
+// * `x` - The number to calculate tan(x)
+// # Returns
+// * `bool` - true if the result is positive, false if the result is negative
+// * `u64` - tan(x) * 1e8
+// # Example
+// * fast_tan(4500000000) = (true, 100000000)
+fn fast_tan(x: u64) -> (bool, u64) {
```
Comment (LucasLvy) [inline src/math/src/trigonometry.cairo:81]: use `i64`

File: src/math/src/trigonometry.cairo [hunk 3]
```diff
@@ -0,0 +1,88 @@
+// Calculate fast sin(x)
+// Since there is no float in cairo, we multiply every number by 1e8
+// # Arguments
+// * `x` - The number to calculate sin(x)
+// # Returns
+// * `bool` - true if the result is positive, false if the result is negative
+// * `u64` - sin(x) * 1e8
+// # Example
+// * fast_sin(3000000000) = (true, 50000000)
+fn fast_sin(x: u64) -> (bool, u64) {
```
Comment (LucasLvy) [inline src/math/src/trigonometry.cairo:10]: use `i64`

**Review by LucasLvy — CHANGES_REQUESTED** [review]

File: src/math/src/trigonometry.cairo (hunk 2 above)
Comment (Soptq) [reply src/math/src/trigonometry.cairo]: fixed ;)

File: src/math/src/trigonometry.cairo (hunk 1 above)
Comment (Soptq) [reply src/math/src/trigonometry.cairo]: fixed ;)

File: src/math/src/trigonometry.cairo (hunk 3 above)
Comment (Soptq) [reply src/math/src/trigonometry.cairo]: fixed ;)

**Comment by LucasLvy** [comment]
why not i64 everywhere ?

**Comment by Soptq** [comment]
> why not i64 everywhere ?

Thanks @LucasLvy for taking time to review this PR!

It seems that `i64` does not implement `core::traits::Div` and `core::traits::Rem`, which is crucial for the fast trigonometric algorithm. Consequently, I decided to firstly implement `sin(x)` for x > 0, and thereby obtaining `sin(x)`, `cos(x)` and `tan(x)` for x < 0 and x > 0 using symmetry properties.

**Review by LucasLvy — APPROVED** [review]

**Pull request merged**